### ISSUE TITLE
Multiple research domains on project details page

### ISSUE THREAD

**Issue by pherterich** [DMPRoadmap/roadmap#3170]
DMPRoadmap v.3.1.0 introduces a primary research domain. DMPonline users have queried if this could become a repeatable field going forward to cover interdisciplinary research. Probably a question for @briri to judge feasibility.

**Comment by briri** [comment]
Yes, this could be converted over to allow for any number of research domains. We would need to:
- Create a new `plans_research_domains` join table
- Create an upgrade Rake task that moves any old data saved to `plans.research_output_id` into the new table
- Update the UI to allow multiples (see the links on the edit Org details page for an example of possible implementation)
- Update the v1 JSON output which currently places this information into the `Tags` array on the RDA Common standard to work with the new table
- Create a ticket to remove the old `plans.research_domain_id` in a future (after people have had a chance to run the script to migrate existing data)
